scripts/apps/users/controllers/UserListController.js

```diff
@@ -75,14 +75,28 @@ export function UserListController($scope, $location, api, _, session, usersServ
     }
 
     function initCriteria(search, filter) {
-        var query = {};
+        let query = {};
+        const canSeeSupportUsers = usersService.isSupport(session.identity);
+
+        if (!canSeeSupportUsers) {
+            query.is_support = {$ne: true};
+        }
 
         if (search.q) {
             query = {
                 $or: [
-                    {username: {$regex: search.q, $options: '-i'}},
-                    {display_name: {$regex: search.q, $options: '-i'}},
-                    {email: {$regex: search.q, $options: '-i'}},
+                    {
+                        username: {$regex: search.q, $options: '-i'},
+                        is_support: {$ne: !canSeeSupportUsers},
```

Comment:
I think you don't have to add it in `$or`, the query should work in a way that on top level it's AND so it must pass is_support + some of the $or filters